stm32-modules/heater-shaker/include/heater-shaker/motor_task.hpp

```diff
@@ -369,64 +372,83 @@ class MotorTask {
-                static_cast<void>(get_message_queue().try_send(
-                    messages::CheckPlateLockStatusMessage{}));
             }
-        } else {
-            plate_lock_state.status = PlateLockState::IDLE_OPEN;
-            static_cast<void>(
-                task_registry->comms->get_message_queue().try_send(
-                    messages::AcknowledgePrevious{.responding_to_id = msg.id}));
         }
+        static_cast<void>(get_message_queue().try_send(check_state_message));
     }
 
     template <typename Policy>
     auto visit_message(const messages::ClosePlateLockMessage& msg,
                        Policy& policy) -> void {
         static constexpr float ClosePower = 1.0F;
-        if ((!policy.plate_lock_closed_sensor_read()) &&
-            (plate_lock_state.status != PlateLockState::IDLE_CLOSED)) {
+        auto check_state_message = messages::CheckPlateLockStatusMessage{
+            .responding_to_id = msg.id, .from_startup = msg.from_startup};
+        if ((policy.plate_lock_closed_sensor_read()) ||
+            (plate_lock_state.status == PlateLockState::IDLE_CLOSED)) {
+            plate_lock_state.status = PlateLockState::IDLE_CLOSED;
+        } else {
             if ((state.status != State::STOPPED_HOMED) &&
                 (state.status != State::STOPPED_UNKNOWN)) {
-                static_cast<void>(
-                    task_registry->comms->get_message_queue().try_send(
-                        messages::AcknowledgePrevious{
-                            .responding_to_id = msg.id,
-                            .with_error =
-                                errors::ErrorCode::MOTOR_NOT_STOPPED}));
+                check_state_message.with_error =
+                    errors::ErrorCode::MOTOR_NOT_STOPPED;
             } else {
-                from_startup = msg.from_startup ? true : false;
-                cached_plate_lock_id = msg.id;
                 policy.plate_lock_set_power(ClosePower);
                 plate_lock_state.status = PlateLockState::CLOSING;
                 polling_time = 0;
-                static_cast<void>(get_message_queue().try_send(
-                    messages::CheckPlateLockStatusMessage{}));
             }
-        } else {
-            plate_lock_state.status = PlateLockState::IDLE_CLOSED;
+        }
+        static_cast<void>(get_message_queue().try_send(check_state_message));
+    }
+
+    template <typename Policy>
+    auto visit_message(const messages::CheckPlateLockStatusMessage& msg,
+                       Policy& policy) -> void {
+        if (msg.with_error != errors::ErrorCode::NO_ERROR) {
+            static_cast<void>(
+                task_registry->comms->get_message_queue().try_send(
+                    messages::AcknowledgePrevious{
+                        .responding_to_id = msg.responding_to_id,
+                        .with_error = msg.with_error}));
+        } else if ((plate_lock_state.status == PlateLockState::IDLE_CLOSED) ||
```

Comment:
What happens if
- Plate lock status is currently `IDLE_OPEN`
- we get a `ClosePlateLockMessage`
- this fires? Won't this trigger incorrectly because the plate lock status is still IDLE_OPEN? or is there something that's guaranteed to change the state to OPENING in between? If it the interrupt firing when the switch closes, is there a guarantee that happens before the check message is recieved? Maybe we should set the state to OPENING/CLOSING in the `OpenPlateLock` and `ClosePlateLock` message handlers

Reply:
`                policy.plate_lock_set_power(ClosePower);
                plate_lock_state.status = PlateLockState::CLOSING;
                polling_time = 0;
`
`ClosePlateLockMessage` will enter this condition, changing state to `CLOSING`

Reply:
Oh! I completely missed that, my bad.

Reply:
First code segment is lines 408-410. Does that answer your question?